Sources/Tracking/Repository/IdentifyRepository.swift

```diff
@@ -0,0 +1,61 @@
+import Common
+import Foundation
+
+internal protocol IdentifyRepository: AutoMockable {
+    func addOrUpdateCustomer(identifier: String,
+                             email: String?,
+                             onComplete: @escaping (Result<Void, CustomerIOError>) -> Void)
+    func removeCustomer()
+}
+
+internal class CIOIdentifyRepository: IdentifyRepository {
+    private let httpClient: HttpClient
+    private let keyValueStorage: KeyValueStorage
+    private let siteId: String
+
+    /// for testing
+    internal init(httpClient: HttpClient, keyValueStorage: KeyValueStorage, siteId: String) {
+        self.httpClient = httpClient
+        self.keyValueStorage = keyValueStorage
+        self.siteId = siteId
+    }
+
+    init(credentials: SdkCredentials, config: SdkConfig) {
+        self.httpClient = CIOHttpClient(credentials: credentials, config: config)
+        self.siteId = credentials.siteId
+        self.keyValueStorage = DICommon.shared.keyValueStorage
+    }
+
+    func addOrUpdateCustomer(
+        identifier: String,
+        email: String?,
+        onComplete: @escaping (Result<Void, CustomerIOError>) -> Void
+    ) {
+        guard let bodyData = JsonAdapter.toJson(AddUpdateCustomerRequestBody(email: email, anonymousId: nil)) else {
+            return onComplete(Result.failure(.httpError(.noResponse)))
+        }
+
+        let httpRequestParameters = HttpRequestParams(endpoint: .identifyCustomer(identifier: identifier), headers: nil,
+                                                      body: bodyData)
+
+        httpClient
+            .request(httpRequestParameters) { [weak self] result in
+                guard let self = self else { return }
+
+                switch result {
+                case .success:
+                    self.keyValueStorage.setString(siteId: self.siteId, value: identifier, forKey: .identifiedProfileId)
```

Comment:
this then obviously has ripple effects down to how we store them, whether or not we go the struct-identity route, IMO we should be storing any supplied identities as a list, and then we can just take the first of them when performing attribute update calls